src/BuildScriptGenerator/DotNetCore/VersionProviders/DotNetCoreVersionProvider.cs

```diff
@@ -11,15 +11,15 @@ namespace Microsoft.Oryx.BuildScriptGenerator.DotNetCore
 {
     internal class DotNetCoreVersionProvider : IDotNetCoreVersionProvider
     {
-        private readonly BuildScriptGeneratorOptions _cliOptions;
+        private readonly CommonOptions _cliOptions;
```

Comment:
Adding a CommonOptions because we want the version providers work for both detectorOptions and ScriptGeneratorOptions. We could rename it if there has a better name.